src/components/TrackingConsent.vue

```diff
@@ -0,0 +1,467 @@
+                    class="nq-button-s"
+                    @click="denyConsent"
+                    :class="{ inverse: theme === constructor.Themes.DARK }"
+                >{{ text.no }}</button>
+                <button
+                    class="nq-button-s"
+                    @click="allowBrowserData"
+                    :class="{ inverse: theme === constructor.Themes.DARK }"
+                >{{ text.browserOnly }}</button>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import I18nMixin from '../i18n/I18nMixin';
+
+interface Consent {
+    allowsBrowserData?: boolean;
+    allowsUsageData?: boolean;
+}
+/**
+ * TrackinConsent Component - used to setup Matomo tracking in a vue-based project
+ */
+@Component
+class TrackingConsent extends Vue {
+    private static _instances: Set<TrackingConsent> = new Set();
+
+    @Prop({
+        type: Object,
+        default:  () => ({
+            get main() {
+                return I18nMixin.$t(
+                    'TrackingConsent',
+                    'Help Nimiq improve by sharing anonymized usage data. Thank you! ❤️',
+                );
+            },
+            get yes() { return I18nMixin.$t('TrackingConsent', 'Yes'); },
+            get no() { return I18nMixin.$t('TrackingConsent', 'No'); },
+            browserOnly: 'Browser-info only',
+        }),
+        validator: (text) => (
+            text && typeof text === 'object' &&
+            text.no && typeof text.no === 'string' && text.no.length &&
+            text.yes && typeof text.yes === 'string' && text.yes.length &&
+            text.main && typeof text.main === 'string' && text.main.length &&
+            text.browserOnly && typeof text.browserOnly === 'string' && text.browserOnly.length
+        ),
+    })
+    public text: {
+        main: string,
+        yes: string,
+        no: string,
+        browserOnly: string,
+    };
+
+    @Prop({
+        type: String,
+        default: 'light',
+        validator: (theme) => Object.values(TrackingConsent.Themes).includes(theme),
+    })
+    public theme: string;
+
+    /** API reference: https://developer.matomo.org/guides/tracking-javascript#configuration-of-the-tracker-object */
+    @Prop({
+        type: Object,
+        default: () => ({
+            setSiteId: 1,
+            setTrackerUrl: TrackingConsent.DEFAULT_TRACKER_URL,
+        }),
+    })
+    public options: {
+        setSiteId: number,
+        setTrackerUrl?: string
+        [k: string]: null | boolean | number | string | string[],
+    };
+
+    @Prop({
+        type: String,
+        default: () => TrackingConsent.DEFAULT_TRACKING_URL,
+    })
+    public trackingScriptUrl: string;
+
+    @Prop({
+        type: Boolean,
+        default: true,
+    })
+    public uiAllowed: boolean;
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public ignoreDoNotTrack: boolean;
```

Comment:
Is it a good idea to make this a setting in the first place? Shouldn't we respect this setting?

Reply:
We do respect this setting by default. But if someone want to not respect it, then he's free to do so. (see Daniel's comment https://github.com/nimiq/vue-components/pull/56#discussion_r334296302)